File: guides/building-a-rest-api-spring-boot-vs-micronaut-implemeting-get/micronautframeworkjacksondatabind/java/src/main/java/example/micronaut/SaasSubscriptionController.java
package example.micronaut;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.PathVariable;
@Controller("/subscriptions") // <1>
class SaasSubscriptionController {
    @Get("/{id}") // <2>
    SaasSubscription findById(@PathVariable Long id) { // <3>
        if (id.equals(99L)) {
            return new SaasSubscription(99L, "Advanced", 2900);
        }
        return null; // <4>
    }
}

File: guides/building-a-rest-api-spring-boot-vs-micronaut-implemeting-get/micronautframeworkjacksondatabind/kotlin/src/main/kotlin/example/micronaut/SaasSubscriptionController.kt
package example.micronaut

import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.PathVariable

@Controller("/subscriptions") // <1>
class SaasSubscriptionController {

    @Get("/{id}") // <2>
    fun findById(@PathVariable id: Long): SaasSubscription? { // <3>
        if (id == 99L) {
            return SaasSubscription(99L, "Advanced", 2900)
        }
        return null // <4>
    }
}
